src/k8s/cmd/k8s/k8s_bootstrap.go

```diff
@@ -31,6 +31,13 @@ var (
 				return fmt.Errorf("failed to create client: %w", err)
 			}
 
+			if c.IsBootstrapped(cmd.Context()) {
+				fmt.Println("Node is already bootstrapped.")
+				// This is not an error as we "archived" the
+				// goal to bootstrap the cluster on this node.
```

Comment:
do you mean `achieved`? also, single-line comments are fine here, no need to keep 80 chars

Reply:
the more i think about it, i would prefer if `k8s bootstrap` fails on a cluster that has already started. `bootstrap` is a one-time action, not a desired state.

Reply:
is there any use-case we would like to accomodate by allow this?

Reply:
I'd say "bootstrapped" is a desired state. 
With `k8s bootstrap` the user wants to achieve something -> bootstrapping a cluster. I don't think that this command should fail if the cluster is already bootstrapped - this is the expected outcome for the user.

Also, we basically offload the check for a bootstrapped cluster to the user. How does the user know if the cluster is bootstrapped - he kicks some APIs and checks for results. This may or may not be sufficient (the API call can fail for other reasons). We as the devs can provide a more robust implementation for this.
See, e.g. the [k8s charm](https://github.com/canonical/k8s-operator/blob/main/charms/k8s/src/charm.py#L89).

Reply:
resolved offline - not returning an error can get messy (e.g. if a joined node is bootstrapped or if the init config is changed)